dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataTablesValidatorTaskTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.teradata;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+import com.google.common.io.ByteSink;
+import com.google.edwmigration.dumper.application.dumper.MetadataDumperUsageException;
+import com.google.edwmigration.dumper.application.dumper.handle.JdbcHandle;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TeradataTablesValidatorTaskTest {
+  @Mock private TaskRunContext context;
+
+  @Mock private JdbcHandle jdbcHandle;
+
+  @Mock private ByteSink sink;
+
+  @Mock private Connection connection;
+
+  @Mock private PreparedStatement statement;
+
+  @Rule public ExpectedException expectedException = ExpectedException.none();
+
+  @Test
+  public void tablesExists() throws Exception {
```

Comment:
Consider adding the expected outcome to the test names, even if the outcome is simple:
`tablesExists_success`, `someTablesDoNotExist_throwsUsageException`.

Reply:
updated